media/js/base/core-datalayer-init.js

```diff
@@ -4,14 +4,31 @@
 
 // init core dataLayer object and push into dataLayer
 $(function() {
-    if (Mozilla && Mozilla.Analytics) {
-        var dataLayer = window.dataLayer = window.dataLayer || [];
+    var dataLayer = window.dataLayer = window.dataLayer || [];
 
-        dataLayer.push({
+    function sendCoreDataLayer() {
+        var dataLayerCore = {
             'event': 'core-datalayer-loaded',
-            'pageId': Mozilla.Analytics.getPageId()
-        });
+            'pageId': Mozilla.Analytics.getPageId(),
+            'pageHasDownload': Mozilla.Analytics.pageHasDownload(),
+            'pageHasVideo': Mozilla.Analytics.pageHasVideo(),
+            'pageVersion': Mozilla.Analytics.getPageVersion(),
+            'testPilotUser': window.navigator.testpilotAddon ? window.navigator.testpilotAddon : 'false'
+        };
+
+        dataLayer.push(dataLayerCore);
     }
 
-    Mozilla.Analytics.updateDataLayerPush();
+    if (Mozilla && Mozilla.Analytics) {
+        if (Mozilla.Client.isFirefox) {
```

Comment:
Here we should use `client.isFirefoxDesktop || client.isFirefoxAndroid` instead, as Firefox for iOS doesn't have version  information available in the same way other platforms do. You could also use a local reference to `Mozilla.Client` using `var client = window.Mozilla.Client;`